pmlc/ast/ast_ops.cc

```diff
@@ -133,13 +135,41 @@ struct GatherOp : Intrinsic {
           "to be a constant float");
     }
 
-    TensorShape shape{tensor.elementType};
-    for (auto i = 0; i < axis.getValue(); i++) {
-      shape.sizes.push_back(tensor.sizes[i]);
+    auto mode = getIntegerValue(evaluator, operands[6]);
+    if (!mode) {
+      throw std::runtime_error("'gather' primitive expects the 'mode' argument "
+                               "to be a constant integer");
+    } else if (mode.getValue() == 1) {
+      if (isDataTypeFloat(idxs.elementType)) {
```

Comment:
I think it is better to add a function `isDataTypeInteger` here.

Reply:
Does this mean we have to take `i1` `si8` `ui8` etc into account even though indices may not have such types?

Reply:
If we are not supporting all integer types like i1, then we should check all specific integer types that we support.

Reply:
I think we support i1, si8 and ui8, no?

Reply:
Yes. What I mean is that `indices` is not likely to have `i1` type so if users pass in `i1` typed `indices` tensor there should be errors thrown. Anyway, we can do a overall check using `isDataTypeInteger` with all the integer types included.

Reply:
I see what you mean. I am not sure if using i1 for indices is valid or not. But if you prefer a subset of integers as valid indices, maybe you can change the function name to something like `isValidIntegerIndices` which check a subset of integers.

Reply:
Maybe keeping the full set of integer types is better because I found no explicit statements about which integer type is not supported by indices.

Reply:
Ok, I think that is good for now.